src/model/ResNet32.py

```diff
@@ -0,0 +1,65 @@
+
+import pytorch_lightning as pl
+import time
+from torch import nn
+
+class LambdaLayer(nn.Module):
+    def __init__(self, lambd):
+        super(LambdaLayer, self).__init__()
+        self.lambd = lambd
+
+    def forward(self, x):
+        return self.lambd(x)
+
+class BasicBlock(nn.Module):
+    def __init__(self, in_planes, planes, stride=1):
+        super(BasicBlock, self).__init__()
+        self.conv1 = nn.Conv2d(in_planes, planes, kernel_size=3, stride=stride, padding=1, bias=False)
+        self.bn1 = nn.BatchNorm2d(planes)
+        self.conv2 = nn.Conv2d(planes, planes, kernel_size=3, stride=1, padding=1, bias=False)
+        self.bn2 = nn.BatchNorm2d(planes)
+        self.shortcut = nn.Sequential()
+        if stride != 1 or in_planes != planes:
+            self.shortcut = LambdaLayer(lambda x:
+                                        nn.functional.pad(x[:, :, ::2, ::2], (0, 0, 0, 0, planes // 4, planes // 4),
+                                                          "constant",
+                                                          0))
+
+    def forward(self, x):
+        out = nn.functional.relu(self.bn1(self.conv1(x)))
+        out = self.bn2(self.conv2(out))
+        out += self.shortcut(x)
+        out = nn.functional.relu(out)
+        return out
+
+class ResNet32(nn.Module):
```

Comment:
Specifying the depth here is Ok, but later on we will make the depth a variable as well, since there is ResNet-56 or ResNet152